src/object.c

```diff
@@ -177,15 +182,18 @@ robj *createStringObjectFromLongLongWithOptions(long long value, int valueobj) {
 /* Wrapper for createStringObjectFromLongLongWithOptions() always demanding
  * to create a shared object if possible. */
 robj *createStringObjectFromLongLong(long long value) {
-    return createStringObjectFromLongLongWithOptions(value,0);
+    return createStringObjectFromLongLongWithOptions(value, LL2STROBJ_AUTO);
 }
 
 /* Wrapper for createStringObjectFromLongLongWithOptions() avoiding a shared
  * object when LFU/LRU info are needed, that is, when the object is used
  * as a value in the key space, and Redis is configured to evict based on
  * LFU/LRU. */
 robj *createStringObjectFromLongLongForValue(long long value) {
-    return createStringObjectFromLongLongWithOptions(value,1);
+    return createStringObjectFromLongLongWithOptions(value, LL2STROBJ_VALUE);
+}
+robj *createll2StringObject(long long value) {
```

Comment:
not sure if that name helps enough to describe what it does, or more importantly assist you in choosing between the various similar functions.
maybe it should be called createStringObjectFromLongLong**WithSds**()?

i.e. we have one function that creates a string object when you don't have any concerns.
one to create string objects that are used as top level values (must have their own LRU)
and now one that must be raw (sds) encoded and never be integer encoded (the EMB part isn't really a concern).